Changes

```diff
@@ -178,6 +178,9 @@ Working version
   (Vincent Laviron, report by François Pottier, review by Nathanaëlle Courant
   and Gabriel Scherer)
 
+- #12037: Fix some data races by using volatile when necessary
```

Comment:
This is the wrong PR number (I noticed that while rebasing one of my recent PRs). @Octachron since this has been cherry-picked to 5.1 you might want to fix it there too.